src/createServer.js

```diff
@@ -1,3 +1,46 @@
 // Write code here
 // Also, you can create additional files in the src folder
 // and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase/convertToCase');
+const { errorHandler } = require('./errorHandler');
+
+function createServer() {
+  const server = http.createServer(async (req, res) => {
+    res.setHeader('Content-type', 'application/json');
+
+    const normalizedURL = new URL(req.url, 'http://localhost:5700');
+
+    const originalText = normalizedURL.pathname.slice(1);
+    const targetCase = normalizedURL.searchParams.get('toCase');
```

Comment:
The `targetCase` parameter should be validated to ensure it is one of the supported cases. If it is not, an error should be thrown or handled appropriately. This is crucial to meet the task's validation requirements.